### ISSUE TITLE
SequentialLabels generates a wrong mapping

### ISSUE THREAD

**Issue by iamSmallY** [fepegar/torchio#840]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug summary

> For example, if a label map has 6 labels with IDs (3, 5, 9, 15, 16, 23), then this will apply a RemapLabels transform with remapping={3: 1, 5: 2, 9: 3, 15: 4, 16: 5, 23: 6}. 

The documentation describes that the behavior of SequentialLabels is to map (3, 5, 9, 15, 16, 23) to (1, 2, 3, 4, 5, 6). But when I use SequentialLabels, I get (3, 1, 2, 3, 4, 5)

And I have read the source code, `SequentialLabels` uses the code below to generate mapping.

```python
unique_labels = torch.unique(image.data)
remapping = {
     unique_labels[i].item(): i
     for i in range(1, len(unique_labels))
}
```

When we input the (3, 5, 9, 15, 16, 23), the mapping is {5: 1, 9: 2, 15: 3, 16: 4, 23: 5}, so the 3 is reused.

### Code for reproduction

```python
import torch
import torchio

subject = torchio.Subject(label=torchio.LabelMap(tensor=torch.Tensor([[[[3, 5, 9, 15, 16, 23]]]])))
transform = torchio.SequentialLabels()
subject = transform(subject)
print(subject["label"][torchio.DATA])
```

### Actual outcome

tensor([[[[3., 1., 2., 3., 4., 5.]]]])

### Expected outcome

tensor([[[[1., 2., 3., 4., 5., 6.]]]])

### System info

```shell
Platform:   Windows-10-10.0.19044-SP0
TorchIO:    0.18.74
PyTorch:    1.10.2+cu102
SimpleITK:  2.1.1 (ITK 5.2)
NumPy:      1.22.2
Python:     3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
```

**Comment by fepegar** [comment]
I might be wrong, but I think your example is inconsistent which the implementation. Wouldn't the expected output be `tensor([[[[0., 1., 2., 3., 4., 5.]]]])`?

**Comment by iamSmallY** [comment]
I also think the output should be `tensor([[[[0., 1., 2., 3., 4., 5.]]]])`, I use `tensor([[[[1., 1., 2 ., 3., 4., 5.]]]])` just because it was written in the original documentation. And in my pr, I modified the document.
